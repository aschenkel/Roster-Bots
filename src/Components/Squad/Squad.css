.Squad-Content{
    -webkit-column-count: 2;
    -moz-column-count: 2;
    column-count: 2;
    margin: 0 12%;
    column-gap: 80px;
    transform: translateZ(0);
  -webkit-margin-top-collapse:discard
}

h1{
    text-align: left;
    margin: 24px;
    color: #bbbbbb; 
	text-shadow: 0px 1px 0px rgba(255,255,255,.3), 0px -1px 0px rgba(0,0,0,.7);
}


@media only screen and (max-width: 900px) {
    .Squad-Content{
        -webkit-column-count: 1;
        -moz-column-count: 1;
        column-count: 1;

    }
}


@media only screen and (max-width: 480px) {
    .Squad-Content{
        -webkit-margin-start: auto;
         -webkit-column-count: 1;
        -moz-column-count: 1;
        column-count: 1;
    }
}



