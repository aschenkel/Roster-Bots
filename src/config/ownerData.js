export const ownerName = "Axel"
export const ownerLastName = "Schenkelman"
export const ownerMail = "axel.schenkelman"
export const ownerGithub = "aschenkel"
