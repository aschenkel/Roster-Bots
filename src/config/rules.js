export const maxScoreAllowed = 100
export const minScoreAllowed = 0
export const numberOfPlayersPerTeam = 15
