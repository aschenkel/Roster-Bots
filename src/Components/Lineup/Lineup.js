import React, { Component } from 'react';
import './Lineup.css';
import Squad from '../Squad/Squad'
import {pickLineUp} from '../../models/Lineup/Lineup'

class Lineup extends Component {
  constructor(props){
    super(props)
    this.state = pickLineUp(this.props.roster)
  }
  render() {
    return (
      <div className="Lineup">
        <Squad starter players={this.state.starters}/>
        <Squad players={this.state.subs}/>
      </div>
    );
  }
}

export default Lineup;