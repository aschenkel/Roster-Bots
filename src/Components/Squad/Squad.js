import React, { Component } from 'react';
import './Squad.css';

const Squad = (props) => {
    console.log(props)
    return (
        //Props = starter = bool.. players = array
        //check if starter
        //map over players -> element -> PlayerCard
      <div className="Squad">
        {props.players.map(player =>{
               <p> player.name</p>
            })}
      </div>
    );
}

export default Squad;