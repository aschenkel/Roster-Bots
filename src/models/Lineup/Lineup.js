import {numberOfStarters, numberOfPlayersPerTeam} from '../../settings/rules'

export const pickLineUp = roster =>  {
    var rosterOrderBySpeed = roster.sort(compareBySpeed);
    var starters = pickStarters(rosterOrderBySpeed.reverse());
    var subs = pickSubs(rosterOrderBySpeed);
    return {starters,subs}
}

const pickStarters = (roster) =>{
    var starters = roster.slice(0, numberOfStarters);
    return starters
}

const pickSubs = (roster) =>{
    var subs = roster.slice(numberOfStarters, numberOfPlayersPerTeam);
    return subs
}


const compareBySpeed = (playerA,playerB) =>{
  if (playerA.speed < playerB.speed)
    return -1;
  if (playerA.speed > playerB.speed)
    return 1;
  return 0;
}

