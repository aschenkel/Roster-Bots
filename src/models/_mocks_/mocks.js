import {numberOfPlayersPerTeam,maxScoreAllowed, numberOfStarters} from '../../settings/rules'


export const createMockPlayer = (score) =>{
    var halfScore = score/2
    return {
        name:mockNameCreator(),
        speed: Math.round(halfScore),
        strength:Math.floor(halfScore),
        score
    }
}

export const mockNameCreator = () => {
    return '081813adc75140e6a12b04f61968607b'
}

export const createMockRoster = () =>{
    var mockPlayers = []
    for (var i = 0; i < numberOfPlayersPerTeam; i++) {
      var properScore = maxScoreAllowed - i;
      mockPlayers.push(createMockPlayer(properScore))
    }
    return mockPlayers
}


