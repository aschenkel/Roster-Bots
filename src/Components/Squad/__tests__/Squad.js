import React from 'react';
import Squad from '../Squad';
import renderer from 'react-test-renderer';
import {createMockRoster} from '../../../models/_mocks_/mocks'
import {pickLineUp} from '../../../models/Lineup/Lineup'
import {numberOfStarters, numberOfPlayersPerTeam} from '../../../config/rules'


var mockRoster = createMockRoster()
var{starters,subs} = pickLineUp(mockRoster)

it('should render starter squad', () => {
    const tree = renderer.create(
        <Squad starter players={starters}/>
        ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render subs squad', () => {
    const tree = renderer.create(
        <Squad players={subs}/>
        ).toJSON();
    expect(tree).toMatchSnapshot();
});

