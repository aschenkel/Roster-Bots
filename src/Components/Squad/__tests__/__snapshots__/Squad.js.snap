// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render starter squad 1`] = `
<div>
  <h1>
    STARTERS
     
  </h1>
  <div
    className="Squad-Content"
  >
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              50
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              50
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          100
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              50
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              49
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          99
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              49
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              49
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          98
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              49
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              48
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          97
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              48
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              48
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          96
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              48
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              47
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          95
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              47
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              47
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          94
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              47
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              46
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          93
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              46
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              46
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          92
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              46
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              45
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          91
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render subs squad 1`] = `
<div>
  <h1>
    SUBS
     
  </h1>
  <div
    className="Squad-Content"
  >
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              45
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              45
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          90
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              45
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              44
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          89
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              44
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              44
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          88
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              44
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              43
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          87
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="data"
      >
        <div
          className="name-container"
        >
          <div
            className="content"
          >
            <p
              className="name-tag"
            >
              name
            </p>
            <p
              className="name"
            >
              081813ad-c751-40e6-a12b-04f61968607b
            </p>
          </div>
        </div>
        <div
          className="atributes"
        >
          <div>
            <p
              className="badge"
            >
              43
            </p>
            <p
              className="atributes-text"
            >
              speed
            </p>
          </div>
          <div>
            <p
              className="badge"
            >
              43
            </p>
            <p
              className="atributes-text"
            >
              strength
            </p>
          </div>
        </div>
      </div>
      <div
        className="player-score"
      >
        <p
          className="score"
        >
           
          86
        </p>
      </div>
    </div>
  </div>
</div>
`;
